test/e2e/apps/job.go

```diff
@@ -466,6 +467,41 @@ done`}
 		gomega.Expect(gotIndexes).To(gomega.Equal(wantIndexes), "expected completed indexes %s, but got %s", wantIndexes, gotIndexes)
 	})
 
+	/*
+		  Release: v1.32
+			Testname: Ensure Pods of an Indexed Job get a unique index for PodIndexLabel key.
+			Description: Create an Indexed job. Job MUST complete successfully.
+			Ensure that created pods have completion index label.
+	*/
+	framework.It("should create pods with completion indexes for an Indexed Job", feature.PodIndexLabel, func(ctx context.Context) {
```

Comment:
After this test settles, when you consider moving this to conformance, I'd squash this with this one: https://github.com/kubernetes/kubernetes/blob/d9b95ea94f41b942440adbdfdbe35fb78a2ed6f7/test/e2e/apps/job.go#L437-L469
because both are almost identical, the only difference is one is checking annotation and this one labels, which both should match.

Reply:
Can you add a TODO here, so that we don't forget to do that?